casanovo/denovo/model_runner.py

```diff
@@ -23,307 +25,350 @@
-    val_is_index = any(
-        [os.path.splitext(fn)[1] in (".h5", ".hdf5") for fn in val_filenames]
-    )
-    if val_is_index and len(val_filenames) > 1:
-        logger.error("Multiple validation HDF5 spectrum indexes specified")
-        raise ValueError("Multiple validation HDF5 spectrum indexes specified")
-    tmp_dir = tempfile.TemporaryDirectory()
-    if train_is_index:
-        train_idx_fn, train_filenames = train_filenames[0], None
-    else:
-        train_idx_fn = os.path.join(tmp_dir.name, f"{uuid.uuid4().hex}.hdf5")
-    valid_charge = np.arange(1, config["max_charge"] + 1)
-    train_index = AnnotatedSpectrumIndex(
-        train_idx_fn, train_filenames, valid_charge=valid_charge
-    )
-    if val_is_index:
-        val_idx_fn, val_filenames = val_filenames[0], None
-    else:
-        val_idx_fn = os.path.join(tmp_dir.name, f"{uuid.uuid4().hex}.hdf5")
-    val_index = AnnotatedSpectrumIndex(
-        val_idx_fn, val_filenames, valid_charge=valid_charge
-    )
-    # Initialize the data loaders.
-    dataloader_params = dict(
-        batch_size=config["train_batch_size"],
-        n_peaks=config["n_peaks"],
-        min_mz=config["min_mz"],
-        max_mz=config["max_mz"],
-        min_intensity=config["min_intensity"],
-        remove_precursor_tol=config["remove_precursor_tol"],
-        n_workers=config["n_workers"],
-    )
-    train_loader = DeNovoDataModule(
-        train_index=train_index, **dataloader_params
-    )
-    train_loader.setup()
-    val_loader = DeNovoDataModule(valid_index=val_index, **dataloader_params)
-    val_loader.setup()
-    # Initialize the model.
-    model_params = dict(
-        dim_model=config["dim_model"],
-        n_head=config["n_head"],
-        dim_feedforward=config["dim_feedforward"],
-        n_layers=config["n_layers"],
-        dropout=config["dropout"],
-        dim_intensity=config["dim_intensity"],
-        custom_encoder=config["custom_encoder"],
-        max_length=config["max_length"],
-        residues=config["residues"],
-        max_charge=config["max_charge"],
-        precursor_mass_tol=config["precursor_mass_tol"],
-        isotope_error_range=config["isotope_error_range"],
-        n_beams=config["n_beams"],
-        top_match=config["top_match"],
-        n_log=config["n_log"],
-        tb_summarywriter=config["tb_summarywriter"],
-        warmup_iters=config["warmup_iters"],
-        max_iters=config["max_iters"],
-        lr=config["learning_rate"],
-        weight_decay=config["weight_decay"],
-    )
-    if config["train_from_scratch"]:
-        model = Spec2Pep(**model_params)
-    else:
-        if not os.path.isfile(model_filename):
+        if train and any(from_scratch):
+            self.model = Spec2Pep(**model_params)
+            return
+        elif self.model_filename is None:
+            logger.error("A model file must be proided")
+            raise ValueError("A model file must be provided")
+
+        if not self.model_filename.exists():
```

Comment:
**issue:** I think that `self.model_filename` is a string, so `exists()` is not a valid function.

Reply:
Good catch! I've added a unit test covering this as well.